### PR TITLE
Add macro to read value from register

### PR DESCRIPTION
**Pull request by bigdinotech**

### CODE REVIEW

**Comment by bigdinotech** [comment]
@eriknyquist @SidLeung Ready for review

File: system/libarc32_arduino101/common/scss_registers.h [hunk 1]
```diff
@@ -38,6 +38,8 @@
 #define MMIO_REG_ADDR(addr) ((volatile uint32_t *)(addr))
 #define MMIO_REG_VAL_FROM_BASE(base, offset) \
     (*((volatile uint32_t *)((base)+(offset))))
+    
+#define REG_VAL(addr) (*(uint32_t *)(addr))
```
Comment (eriknyquist) [inline system/libarc32_arduino101/common/scss_registers.h]: Why do we need this? we already have MMIO_REG_VAL. Line 37.

Comment (bigdinotech) [inline system/libarc32_arduino101/common/scss_registers.h]: It was not necessary to use volatile for all my register accesses in the Power Management library. I could just make everything volatile which should also work.

Comment (eriknyquist) [inline system/libarc32_arduino101/common/scss_registers.h]: No need to make anything volatile-- MMIO_REG_VAL casts the passed pointer as a volatile, which forces volatile semantics on whatever that pointer happens to point to.

And, as was reinforced last week during our course-- you always use volatile to access memory mapped HW! So always using MMIO_REG_VAL is a good idea.

**Pull request closed without merging**